core/testing/router.go

```diff
@@ -0,0 +1,159 @@
+package coretesting
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	appmodulev2 "cosmossdk.io/core/appmodule/v2"
+	"cosmossdk.io/core/router"
+	"cosmossdk.io/core/transaction"
+)
+
+var ErrNoHandler = errors.New("no handler")
+
+// NewMsgRouterBuilder is a router that routes messages to their respective handlers.
+func NewMsgRouterBuilder() *ReflectionRouterBuilder {
```

Comment:
Core-only router builder that uses `reflect` instead of protoreflect for test pathfinding